scripts/cost_metrics_qa.py

```diff
@@ -198,42 +293,93 @@ def main():
     )
 
-    print(f"\nProcess complete for {CM_PATH.name}")
+    print(f"\nProcess complete for {filename}")
     print("_" * 75)
 
 
 # # # # PROCESS # # # #
 
 if __name__ == "__main__":
-
-    # If a single costs pathlib.Path is provided - just run main for that
-    if isinstance(CM_PATH, pathlib.Path):
-        main()
-
-    # If path has been provided as a raw string, convert to pathlib.Path
-    elif isinstance(CM_PATH, str):
-        CM_PATH = pathlib.Path(CM_PATH)
-        main()
-
-    # Otherwise, if a list of paths are provided, run main on all paths.
-    elif isinstance(CM_PATH, list):
-        ALL_PATHS = CM_PATH
-        print(f"Detected {len(CM_PATH):,} cost files. Processing iteratively.")
-
-        for CM_PATH in ALL_PATHS:
-            if not isinstance(CM_PATH, pathlib.Path):
-                CM_PATH = pathlib.Path(CM_PATH)
-            main()
+    if DB_QA:
+        # Create DB connection URL
+        conn_url = create_db_url(
+            username=DB_USERNAME,
+            password=DB_PASSWORD,
+            host=DB_HOST,
+            port=DB_PORT,
+            database_name=DB_NAME,
+            engine="pg8000",
+        )
+        print(f"Using DB url: {conn_url}")
+
+        # Download the dataset
+        data = pull_down_cost_metrics(
+            db_conn_url=conn_url,
+            run_id=DB_RUN_ID,
+            schema=DB_SCHEMA,
+            table=DB_TABLE,
+        )
+        print("Data downloaded")
+
+        # Run QA
+        main(costs_data=data)
+
+    else:  # Must be reading files locally from CM_PATH
+        if not CM_PATH:
+            raise ValueError(
+                "`CM_PATH` is empty. Did you mean to perform QA on DB datasets?"
+            )
+
+        # If a single costs pathlib.Path is provided - just run main for that
+        if isinstance(CM_PATH, pathlib.Path):
+            # Load in data
+            print(f"\nReading costs: {CM_PATH}")
+            data = pd.read_csv(CM_PATH)
+            print("Costs read")
+
+            main(costs_data=data)
+
+        # If path has been provided as a raw string, convert to pathlib.Path
+        elif isinstance(CM_PATH, str):
+            CM_PATH = pathlib.Path(CM_PATH)
+
+            # Load in data
+            print(f"\nReading costs: {CM_PATH}")
+            data = pd.read_csv(CM_PATH)
+            print("Costs read")
+
+            main(costs_data=data)
+
+        # Otherwise, if a list of paths are provided, run main on all paths.
+        elif isinstance(CM_PATH, list):
+            ALL_PATHS = CM_PATH
+            print(f"Detected {len(CM_PATH):,} cost files. Processing iteratively.")
+
+            for CM_PATH in ALL_PATHS:
+                if not isinstance(CM_PATH, pathlib.Path):
+                    CM_PATH = pathlib.Path(CM_PATH)
+
+                # Load in data
+                print(f"\nReading costs: {CM_PATH}")
+                data = pd.read_csv(CM_PATH)
+                print("Costs read")
+
+                main(costs_data=data)
```

Comment:
There's alot of duplicated code here which could easily be avoided one way is to convert a single path into a list with one item then everything can be treated the same. Also, there's no real harm in converting a Path to a Path so you don't need to check if its a string you can just attempt to convert to a Path regardless.
```suggestion
        if isinstance(CM_PATH, (pathlib.Path, str)):
            all_paths = [CM_PATH]
        else:
            # Assume CM_PATH is a list or iterable
            all_paths = CM_PATH

        print(f"Detected {len(all_paths):,} cost files. Processing iteratively.")

        for path in all_paths:
            path = pathlib.Path(path)
            
            # Load in data
            print(f"\nReading costs: {path}")
            data = pd.read_csv(path)
            print("Costs read")

            main(costs_data=data)
```

Another thing, only use all caps for module level constants variables should be all lowercase.